packages/dds/tree/src/test/simple-tree/schemaFactoryRecursive.spec.ts

```diff
@@ -137,6 +137,7 @@ describe("SchemaFactory Recursive methods", () => {
 			type _checkChild = requireTrue<areSafelyAssignable<Child, ObjectRecursive | undefined>>;
 			type Constructor = ConstructorParameters<typeof ObjectRecursive>;
 			type _checkConstructor = requireTrue<
+				// @ts-expect-error Non-optional fields seem to trip up TSC, even if `undefined` is part of the type union. Making the field optional seems to work as expected.
```

Comment:
@CraigMacomber this case now fails, but making the property explicitly optional works (see new case below). It isn't clear to me how vital the non-optional-ness of the first case here is. In practice, object node constructors seem to be working as expected.

Reply:
This builds:

```
		class Foo extends factory.objectRecursive("foo", {
			/**
			 * Test2
			 */
			y: factory.number,
		}) {}
		type _check0 = ValidateRecursiveSchema<typeof Foo>;
		const _check2 = new Foo({});
```

It seems important that we don't make required field optional. Allowing people to omit required data seems worse than forcing them to provide optional data in my opinion. That is a subjective choice, but being stricter is something we can walk back, where being permissive isn't something we can change without it being a breaking change.

Reply:
Given we already have cases where recursive nodes have to be more verbose when constructing them (ex: they are not implicitly constructable), I think not supporting shorter form (omitting optional) for them seems reasonable if we can't find a good way around it.

Reply:
There was a bug in `FieldHasDefaultUnsafe`. That has now been fixed, and everything seems to be working as intended (with the new explicit required recursive field tests in place).